scene/resources/camera_texture.cpp

```diff
@@ -47,6 +47,10 @@ void CameraTexture::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "camera_is_active"), "set_camera_active", "get_camera_active");
 }
 
+void CameraTexture::_on_format_changed() {
+	callable_mp((Resource *)this, &Resource::emit_changed).call_deferred();
```

Comment:
`emit_changed` alreaady handles deferring if it needs. Do you still need to explicitly defer it again here? This might lead to double deferring and two frames of delay.

Reply:
Skipping call_deferred results in some errors so I'm doing it wrong or this is the place where call_deferred is really necessary.
![errors](https://github.com/user-attachments/assets/3fa68c88-3593-48c5-b54d-ea39af826d45)

Reply:
```suggestion
	# FIXME: `emit_changed` is more appropriate, but causes errors for some reason.
	callable_mp((Resource *)this, &Resource::emit_changed).call_deferred();
```

Reply:
For the record, just reposting some comments from @RandomShaper and @smix8 after I asked maintainers about how to do this properly.

> @RandomShaper
> `Resource::emit_changed()` can only work from the main thread or a resource loading thread.
> In the latter case, it employs some deferring magic to keep things fine on the main thread.
> It won't handle being called from an arbitrarty thread, which seems to be the case here.
> I'd like to know why it is being called from another thread in the first place.

> @smix8 
> I think the problem in that PR is more that the `CameraServer.add_feed()` function, that should be called on the main thread, is called directly from the thread function that this PR adds. The base CameraServer also emits a changed signals with that function but without any deferred so that entire function is clearly designed to be only called on the main thread.
> Using the call_deferred() is always a sloppy fix, because as the PR maker already noted, you sometimes need to do it twice because something else might flush the queue before while whatever uses your signal connected node or resource might still be locked. It is not running just once each frame.
> imo the CameraServer should just sync its own updates with the main thread from its own queue, piggybacking on the main message queue with signals is always a valley of tears because that queue gets flushed at random points in the physics, scenetree, and main loop and maybe even in some other cursed places as well. It is totally overloaded with ugly hacks by now.

> @RandomShaper
> `SceneTree`'s `process_frame` signal may be a suitable point to hook to.

> @smix8 
> or do as e.g. the XRServer and just sync/process before all the script loop so things are updated before user scripts try to work with it.

This PR has been waiting long enough so I don't think this should be blocking now, but that's some food for thought and maybe further rework of the implementation to avoid future issues.